Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -1241,15 +1246,15 @@ std::pair<double, double> HGCalDDDConstants::waferPosition(int waferU, int wafer
 int HGCalDDDConstants::waferType(DetId const& id) const {
   int type(1);
   if ((mode_ == HGCalGeometryMode::Hexagon8) || (mode_ == HGCalGeometryMode::Hexagon8Full)) {
-    type = ((id.det() != DetId::Forward) ? (1 + HGCSiliconDetId(id).type()) : (1 + HFNoseDetId(id).type()));
+    type = ((id.det() != DetId::Forward) ? HGCSiliconDetId(id).type() : HFNoseDetId(id).type());
   } else if ((mode_ == HGCalGeometryMode::Hexagon) || (mode_ == HGCalGeometryMode::HexagonFull)) {
-    type = waferTypeL(HGCalDetId(id).wafer());
+    type = waferTypeL(HGCalDetId(id).wafer()) - 1;
   }
   return type;
 }
 
 int HGCalDDDConstants::waferType(int layer, int waferU, int waferV) const {
-  int type(-1);
+  int type(HGCSiliconDetId::HGCalCoarseThick);
```

Comment:
Here your are changing the default value from "-1" to "2". Is that default value ever accessed? Was anything else modified in the rest of the code to take into account this change? (I couldn't easily find it...)

Reply:
-1 does not mean anything - it was used earlier so that all variables are initialized to some value. That default value is never accessed

Reply:
Ok, thank you Sunanda. I just wanted to be sure that such a modified initialization default is never accessed, which is what I expected but not obvious from the code itself.